cognite/neat/rules/models/_base_rules.py

```diff
@@ -210,10 +190,70 @@ def headers_by_sheet(cls, by_alias: bool = False) -> dict[str, list[str]]:
             headers_by_sheet[sheet_name] = [
                 (field.alias or field_name) if by_alias else field_name
                 for field_name, field in model_fields.items()
-                if field_name != "validators_to_skip"
+                if field_name != "validators_to_skip" and not field.exclude
             ]
         return headers_by_sheet
 
+    def dump(
+        self,
+        mode: Literal["python", "json"] = "python",
+        by_alias: bool = False,
+        exclude: IncEx = None,
+        exclude_none: bool = False,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        as_reference: bool = False,
+        entities_exclude_defaults: bool = True,
+    ) -> dict[str, Any]:
+        """Dump the model to a dictionary.
+
+        This is used in the Exporters to dump rules in the required format.
+        """
+        for field_name in self.model_fields.keys():
+            value = getattr(self, field_name)
+            # Ensure deterministic order of properties
+            if isinstance(value, SheetList):
+                value.sort(key=lambda x: x._identifier())
+
+        context: dict[str, Any] = {"as_reference": as_reference}
+        if entities_exclude_defaults:
+            context["metadata"] = self.metadata
+
+        if self.reference is not None:
+            exclude_input: IncEx
+            if isinstance(exclude, dict):
```

Comment:
why `exclude` can be of various types? some explanation would be helpful as a comment or similar

Reply:
This is what pydantic accepts. Adding comment to explain why this is done.